### PR TITLE
Add 1.20.5 / 1.20.6 support

### PR DESCRIPTION
**Pull request by TechnicallyCoded**
Tested to reload a plugin in 1.20.5. Please leave any bugs below.

--- WARNING ---
This hasn't been tested for 1.20.4 or lower although is has been written with such backwards compatibility in mind.

### CODE REVIEW

**Comment by TechnicallyCoded** [comment]
I will note that enough code has changed that it's best to test this version to death instead of trying to read every change. If you want to do so, just review the entire file without the diff mode, basically everything changed in BukkitCommandWrap to make it more future-proof.

**Comment by TheBlackEntity** [comment]
The comments you added are too many and most of them are unnecessary. 

You also have up half way through. 

I suggest removing them completely

**Comment by TechnicallyCoded** [comment]
( For the record, this fixes #35 )

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java [hunk 1]
```diff
@@ -25,139 +31,81 @@ public class BukkitCommandWrap {
...
-            try {
-                this.minecraftServerClass = Class.forName("net.minecraft.server.MinecraftServer");
-            } catch (ClassNotFoundException classNotFoundException) {
-                classNotFoundException.addSuppressed(e);
-                classNotFoundException.printStackTrace();
-                return;
-            }
-        }
+    private static @NotNull String cbPrefix() {
+        return Bukkit.getServer().getClass().getPackage().getName();
```
Comment (TheBlackEntity) [inline src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:51]: I think you forgot something here...

What is the `craftBukkitVersioning` variable for, if we're ignoring it?

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java [hunk 2]
```diff
@@ -25,139 +31,81 @@ public class BukkitCommandWrap {
...
+        if (this.craftBukkitVersioning) {
+            this.nmsVersion = versionPart;
         }
+
+        boolean nmsVers = true;
+        try {
+            Class.forName("net.minecraft.server.MinecraftServer");
+            nmsVers = false;
+        } catch (ClassNotFoundException ignore) {}
+        this.nmsVersioning = nmsVers;
```
Comment (TheBlackEntity) [inline src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:47]: Why do it like this?

Couldn't we just set it to false by default and set it to true, if we encounter the `ClassNotFoundException`?

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java [hunk 3]
```diff
@@ -177,92 +125,193 @@ public void sync() {
...
+        if (!resolveGetServerMethod()) return;
+        Object server = getServerInstance();
+
+        if (!resolveVanillaCommandDispatcherField()) return;
+        Object commandDispatcher = getCommandDispatcher(server);
+
+        if (!resolveBField()) return;
+
+        CommandDispatcher b = getDispatcher(commandDispatcher);
+        if (b == null) return;
```
Comment (TheBlackEntity) [inline src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:129]: I'd like to switch to meaningful names.

Would you be willing to do it for me?

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java [hunk 4]
```diff
@@ -177,92 +125,193 @@ public void sync() {
...
-        } catch (NoSuchFieldException e) {
+    private @Nullable CommandDispatcher getDispatcher(Object commandDispatcher) {
+        CommandDispatcher b = null;
+        try {
+            b = (CommandDispatcher) this.bField.get(commandDispatcher);
+        } catch (IllegalAccessException e) {
             e.printStackTrace();
-            return;
         }
+        return b;
```
Comment (TheBlackEntity) [inline src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:161]: Please move this to line #168 and return null instead.

You should be able to remove line #166 by doing so

Comment (TheBlackEntity) [reply src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:161]: Same thing goes for the other methods, of course

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java [hunk 5]
```diff
@@ -177,92 +125,193 @@ public void sync() {
...
+    private boolean resolveMcServerClass() {
+        if (this.minecraftServerClass == null) {
+            try {
+                this.minecraftServerClass = Class.forName(nmsPrefix("MinecraftServer"));
+            } catch (ClassNotFoundException ex) {
+                ex.printStackTrace();
+                return false;
+            }
+        }
+        return true;
```
Comment (TheBlackEntity) [inline src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:193]: Should we include the return statement in the try block?

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java [hunk 6]
```diff
@@ -1,20 +1,26 @@
 package com.rylinaux.plugman.util;
 
+import com.mojang.brigadier.CommandDispatcher;
 import com.mojang.brigadier.tree.CommandNode;
 import com.mojang.brigadier.tree.RootCommandNode;
 import org.bukkit.Bukkit;
 import org.bukkit.command.Command;
 import org.bukkit.entity.Player;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
```
Comment (TheBlackEntity) [inline src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:10]: Please refrain from using these Annotations.

I don't see a point in using them, if it is not part of some API

**Review by TheBlackEntity — CHANGES_REQUESTED** [review]
Note:
I'm currently in a code style transition, so I may have added comments that might conflict with the code style that is currently being used in this project.

If you think I said something stupid, just tell me xd

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java (hunk 1 above)
Comment (TechnicallyCoded) [reply src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:51]: I don't see what was forgotten. craftBukkitVersioning was for your use if you needed it later but that seems like it was unnecessary so I removed it

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java (hunk 2 above)
Comment (TechnicallyCoded) [reply src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:47]: Shortest / cleanest code, nothing else to the reasoning for me

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java (hunk 3 above)
Comment (TechnicallyCoded) [reply src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:129]: I actually don't know what they are supposed to mean. If you know, I can rename them no problem.

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java (hunk 1 above)
Comment (TheBlackEntity) [reply src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:51]: > I don't see what was forgotten. craftBukkitVersioning was for your use if you needed it later but that seems like it was unnecessary so I removed it

Oh... Yeah, I misunderstood the code

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java (hunk 3 above)
Comment (TheBlackEntity) [reply src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:129]: Well... In this case, you can rename `b` to commandDispatcher xd

But it's fine. I'll do it at some point

File: src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java (hunk 4 above)
Comment (TechnicallyCoded) [reply src/main/java/com/rylinaux/plugman/util/BukkitCommandWrap.java:161]: Updating...

**Comment by TechnicallyCoded** [comment]
Changes were made. Lmk if anything new is needed

**Pull request merged**